src/pudl_archiver/orchestrator.py

```diff
@@ -202,7 +201,8 @@ async def run(self):
                 self._update_dataset_settings(published)
             return published
         else:
-            return self.new_deposition
+            await self.depositor.delete_deposition(self.new_deposition)
+            return self.deposition
```

Comment:
To me it reads like this deletion is happening at the end of the archiver run -- it tries to make a new deposition, and if it finds that nothing has changed, it deletes the draft deposition, meaning that we wouldn't have zombie drafts floating around between updates. That seems like the desired behavior to me.

I might not be understanding how this works, but is it always uploading the new data to a draft, regardless of whether there's been a change? It seems like it should be possible to figure out whether a new version is necessary just based on the list of files and their MD5 checksums -- if it's the same set of files, and they all have the same checksums (the downloaded versions locally compared to the previously uploaded versions on Zenodo) then it seems like creating a new draft deposition and uploading the files could be avoided entirely. The process of uploading the EPA CEMS data to Zenodo was a major source of errors previously (thousands of files, and several GB of data).

Reply:
It's only uploading data if there's something new or something that has changed. When it creates a draft it will automatically contains the contents of the previous version, so we will never re-upload something that hasn't changed. Basically we create a new draft to work off of, and we compare downloaded files to this mirror of the previous version. If nothing needs changing we previously would just leave this draft and come back to it next time. This new change will just clean up after it's self and delete the draft so it doesn't hang around.

Reply:
> To me it reads like this deletion is happening at the end of the archiver run -- it tries to make a new deposition, and if it finds that nothing has changed, it deletes the draft deposition, meaning that we wouldn't have zombie drafts floating around between updates. That seems like the desired behavior to me.

Yeah, that is what is happening.

> I might not be understanding how this works, but is it always uploading the new data to a draft, regardless of whether there's been a change? 

No. It always creates a draft, which is automatically populated with a snapshot of the data. Then the archiver downloads all the data from the source, and creates a changeset based on MD5 comparisons with the draft deposition (see `DepositionOrchestrator._generate_changes`). We then apply that changeset, which may be empty (see `DepositionOrchestrator._apply_changes`).

Reply:
Ah okay, I didn't realize it would just roll the old files over into the new draft deposition. That makes sense since it does file-level deduplication on the backend.